include/swift/IDE/CodeCompletionResultPrinter.h

```diff
@@ -13,12 +13,18 @@
 #ifndef SWIFT_IDE_CODECOMPLETIONRESULTPRINTER_H
 #define SWIFT_IDE_CODECOMPLETIONRESULTPRINTER_H
 
+#include "llvm/ADT/StringRef.h"
 #include "llvm/Support/raw_ostream.h"
+#include "llvm/Support/Allocator.h"
 
 namespace swift {
+
+class NullTerminatedStringRef;
```

Comment:
Is there a specific reason why you don’t just want to 
```cpp
#include "swift/Basic/StringExtras.h`
```
here? I usually try to avoid forward declarations as much as possible because it makes the code fragile if we should rename/remove `NullTerminatedStringRef` in the future.

Reply:
I'm actually the opposite 😅. More includes == slower compiles. What do you mean by "fragile" here?

Reply:
FWIW, I confirmed that Xcode's refactoring picked up these (i.e. forward decl and the usages in this header)